clar2wasm/src/words/arithmetic.rs

```diff
@@ -74,25 +74,39 @@ impl SimpleWord for Sub {
         arg_types: &[TypeSignature],
         return_type: &TypeSignature,
     ) -> Result<(), GeneratorError> {
-        let type_suffix = match return_type {
-            TypeSignature::IntType => "int",
-            TypeSignature::UIntType => "uint",
+        let func = match return_type {
+            TypeSignature::IntType => {
+                let type_suffix = "int";
+                if arg_types.len() == 1 {
+                    // unary 'int' subtraction:
+                    // multiply by -1.
+                    builder.i64_const(-1);
+                    builder.i64_const(-1);
+                    generator.func_by_name(&format!("stdlib.mul-{type_suffix}"))
```

Comment:
Super inefficient, should be `0 - n` instead.

Reply:
Now that I’m thinking about it, it’s also incorrect: `-i128::MIN` should be an overflow error.

Reply:
Changed the implementation to use `0 - n`.